### ISSUE TITLE
auth kwarg cannot be used to authenticate an HTTPS request through a HTTP proxy

### ISSUE THREAD

**Issue by pjjw** [kennethreitz/requests#2003]
i mean so this is probably more of a rant than an issue, but that was 3 hours i won't get back.

was trying to pipe requests through a squid proxy in a lights-out environment. credentials come from a data store separate from configuration, so seemed reasonable to use requests.auth.HTTPProxyAuth(l, p) as the auth kwarg as (what i thought) a supplement to the proxies dict fed to the proxies kwarg. this works fine for http requests, but this seems to be by accident- it falls apart when making an https request through a proxy, when urllib3 uses the CONNECT verb to encapsulate the request.

basically requests.auth.HTTPProxyAuth doesn't make it clear that it's not the correct (always working) way to authenticate to a proxy, but that you must put the credentials into the proxy url in the proxies dict. deprecation/removal of that class would probably make the interface cleaner- failing that, a note in the source would be nice.

**Comment by Lukasa** [comment]
Yeah, agreed: I think we should aim to remove `HTTPProxyAuth` in version 3.0.

**Comment by sigmavirus24** [comment]
:+1: for 3.0 removal.

**Comment by nateprewitt** [comment]
Are we still in agreement that this should be on it's way?

**Comment by sigmavirus24** [comment]
@nateprewitt I am in agreement, but I would be unsurprised if Cory changed his mind. Since he's on holiday, why don't you hold off on working on this so you don't waste your time.

**Comment by nateprewitt** [comment]
Yep, will do. Just floating it out there for when you've both got a free moment. Thanks @sigmavirus24 :)

**Comment by Lukasa** [comment]
My mind isn't changed at this time. Let's kill it. =D

**Comment by nateprewitt** [comment]
This will be resolved in 3.0.0 with #3773.
